common/lending_iterator/kmerge.rs

```diff
@@ -92,13 +92,24 @@ where
     fn seek(&mut self, key: &K) {
         self.iterators = mem::take(&mut self.iterators)
             .drain()
-            .map(|mut it| {
-                it.iter.seek(key);
-                it
+            .filter_map(|mut it| {
+                it.iter.peek();
+                if let Some(item) = it.iter.get_peeked() {
+                    if it.iter.compare_key(item, key) == Ordering::Less {
+                        it.iter.seek(key);
+                    }
+                }
+                it.iter.peek().is_some().then_some(it)
             })
```

Comment:
Filtering out empty iterators in the merge (otherwise the comparison panics during heap sifts), and only seeking where it's valid to do so.